Add two howtos related to the migration to Semgrepignore v2 (#2012)
semgrep/semgrep-docs: docs/kb/semgrep-code/semgrepignore-ignored.md

Starting with Semgrep CE 1.112.0, the Semgrepignore specification has
changed slightly to better align with Git and Gitignore and to offer
more flexibility.
The new specification is referred to as
[Semgrepignore v2](/docs/semgrepignore-v2-reference).
If you're getting findings in files that should have been
ignored according to your `.semgrepignore` file, check the
following:

1. If you're using Git, check that the `.semgrepignore` file is at the
   root of the Git project or at least is within the project.
   `.semgrepignore` files can be placed in any folder in the project
   and follow the same specification as `.gitignore` files,
   which they extend.
2. If you're not using Git, check that the `.semgrepignore` file
   is in the folder passed on the `semgrep scan` command line.
   For example, if the command is `semgrep scan foo/`, you must move
   the `.semgrepignore` file from the current folder
   to `foo/.semgrepignore`.

To ensure you're using Semgrepignore v2, pass the flag
`--semgrepignore-v2` to `semgrep scan` or to `semgrep
ci`. To use the legacy Semgrepignore v1 implementation, use
`--no-semgrepignore-v2`. These options are for troubleshooting the

> migration from v1 to v2. They will disappear soon as v2 becomes the

Reviewer: ```suggestion
migration from v1 to v2. These flags will be removed when v2 becomes
```

only implementation available.

## Best practices

* When scanning a whole project, run `semgrep` from the project root.
* Place a `.semgrepignore` file at the project root.

> * Optionally, place `.semgrepignore` files in subfolders so as to keep the

Reviewer: The plural implies that semgrepignore files are additive?

Reviewer: Yep seems that v2 allows multiple additive semgrepignore files.
